# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/jsactor/JSActorManager.cpp
# Commit: 7229926a45c2
# Full Hash: 7229926a45c2e43136c7f8f597a92776658edbd5
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-10-14 09:52:12
# Description:
#   Bug 1660539 - Make StructuredCloneData args optional in JSActor RawMessage, r=kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93122
# ==============================================================================

diff -r 1aa2dc4b280e -r 7229926a45c2 dom/ipc/jsactor/JSActorManager.cpp
--- a/dom/ipc/jsactor/JSActorManager.cpp	Tue Oct 13 16:57:32 2020 +0000
+++ b/dom/ipc/jsactor/JSActorManager.cpp	Tue Oct 13 17:43:56 2020 +0000
@@ -107,9 +107,10 @@
     }                                      \
   } while (0)
 
-void JSActorManager::ReceiveRawMessage(const JSActorMessageMeta& aMetadata,
-                                       ipc::StructuredCloneData&& aData,
-                                       ipc::StructuredCloneData&& aStack) {
+void JSActorManager::ReceiveRawMessage(
+    const JSActorMessageMeta& aMetadata,
+    Maybe<ipc::StructuredCloneData>&& aData,
+    Maybe<ipc::StructuredCloneData>&& aStack) {
   MOZ_ASSERT(nsContentUtils::IsSafeToRunScript());
 
   CrashReporter::AutoAnnotateCrashReport autoActorName(
@@ -134,9 +135,13 @@
   Maybe<JS::AutoSetAsyncStackForNewCalls> stackSetter;
   {
     JS::Rooted<JS::Value> stackVal(cx);
-    aStack.Read(cx, &stackVal, error);
-    if (error.Failed()) {
-      return;
+    if (aStack) {
+      aStack->Read(cx, &stackVal, error);
+      if (error.Failed()) {
+        error.SuppressException();
+        JS_ClearPendingException(cx);
+        stackVal.setUndefined();
+      }
     }
 
     if (stackVal.isObject()) {
@@ -156,10 +161,12 @@
   }
 
   JS::Rooted<JS::Value> data(cx);
-  aData.Read(cx, &data, error);
-  if (error.Failed()) {
-    CHILD_DIAGNOSTIC_ASSERT(false, "Should not receive non-decodable data");
-    return;
+  if (aData) {
+    aData->Read(cx, &data, error);
+    if (error.Failed()) {
+      CHILD_DIAGNOSTIC_ASSERT(false, "Should not receive non-decodable data");
+      return;
+    }
   }
 
   switch (aMetadata.kind()) {